File: cos/olympus/src/main/java/cos/olympus/game/events/MeleeAttack.java
package cos.olympus.game.events;

import cos.olympus.game.Creature;
import cos.ops.Direction;

public record MeleeAttack(
        @Override int id,
        int tick,
        int x,
        int y,
        Direction dir,
        @Override Creature source
) implements Spell, Event {

}

File: cos/src/main/java/cos/olympus/game/events/MeleeAttack.kt
package cos.olympus.game.events

import cos.olympus.game.Actor
import cos.olympus.game.Pos
import cos.olympus.game.TickId
import cos.ops.Direction

data class MeleeAttack(
    override val id: Int,
    override val tick: TickId,
    val x: Pos,
    val y: Pos,
    val dir: Direction,
    override val source: Actor
) : Spell, Event
